fearless/Common/QRCoding/QRService.swift

```diff
@@ -0,0 +1,50 @@
+import RobinHood
+
+protocol QRServiceProtocol: AnyObject {
+    @discardableResult
+    func generate(
+        with qrType: QRType,
+        qrSize: CGSize,
+        runIn queue: DispatchQueue,
+        completionBlock: @escaping (Result<UIImage, Error>?) -> Void
+    ) throws -> Operation
+}
+
+final class QRService {
+    let operationFactory: QROperationFactoryProtocol
+    let operationQueue: OperationQueue
+
+    private let encoder: QREncoderProtocol
+
+    public init(
+        operationFactory: QROperationFactoryProtocol,
+        encoder: QREncoderProtocol = QREncoder(),
+        operationQueue: OperationQueue = OperationQueue()
+    ) {
+        self.operationFactory = operationFactory
+        self.encoder = encoder
+        self.operationQueue = operationQueue
+    }
+}
+
+extension QRService: QRServiceProtocol {
+    @discardableResult
+    func generate(
+        with qrType: QRType,
```

Comment:
you can avoid passing this parameter in case of passing concrete encoder implementation in this class.